File: LSP/Employee.cpp
#include <iostream>

class Employee {
protected:
    std::string name;
    int id;

public:
    Employee(const std::string& name, int id) : name(name), id(id) {}

    virtual double calculateSalary() {
        // Base salary calculation logic
        return 0.0;
    }
};

class Manager : public Employee {
private:
    int teamSize;

public:
    Manager(const std::string& name, int id) : Employee(name, id), teamSize(0) {}

    void setTeamSize(int teamSize) {
        this->teamSize = teamSize;
    }

    double calculateSalary() override {
        // Manager-specific salary calculation logic
        double baseSalary = Employee::calculateSalary();
        double bonus = teamSize * 1000;
        return baseSalary + bonus;
    }
};

class Developer : public Employee {
private:
    std::string language;

public:
    Developer(const std::string& name, int id) : Employee(name, id), language("") {}

    void setLanguage(const std::string& language) {
        this->language = language;
    }

    double calculateSalary() override {
        // Developer-specific salary calculation logic
        double baseSalary = Employee::calculateSalary();
        double bonus = (language == "Cpp") ? 1000 : 500;
        return baseSalary + bonus;
    }
};

File: LSP/Employee.py
class Employee:
    def __init__(self, name, id):
        self.name = name
        self.id = id

    def calculate_salary(self):
        # Base salary calculation logic
        pass


class Manager(Employee):
    def __init__(self, name, id):
        super().__init__(name, id)
        self.team_size = 0

    def set_team_size(self, team_size):
        self.team_size = team_size

    def calculate_salary(self):
        # Manager-specific salary calculation logic
        base_salary = super().calculate_salary()
        bonus = self.team_size * 1000
        return base_salary + bonus


class Developer(Employee):
    def __init__(self, name, id):
        super().__init__(name, id)
        self.language = ""

    def set_language(self, language):
        self.language = language

    def calculate_salary(self):
        # Developer-specific salary calculation logic
        base_salary = super().calculate_salary()
        bonus = 1000 if self.language == "Python" else 500
        return base_salary + bonus
